### PR TITLE
support release work

### PR DESCRIPTION
**Pull request by changev**
**modify travis config**

used

per PR-merge of master build deb and upload them to bintray, named on-tftp_1.1-1master-20160923181601Z.deb .

now

per PR-merge build on-tftp_1.2.3~rc-20160923181601Z-125b07b.deb or on-tftp_1.2.2~devel-20150923181601Z-535b07b.deb with deb_component "main", we call this ci-builds.
per push Tag build on-tftp_1.2.3.deb with deb_component "release", we call this release build
This feature will be applied on every repo in the future.
@RackHD/corecommitters @panpan0000 @PengTian0

### CODE REVIEW

**Comment by rackhdmirror** [comment]
Build finished.

File: debian/changelog [hunk 1]
```diff
@@ -1,4 +1,4 @@
-on-tftp (1.1-1) unstable; urgency=low
+on-tftp (1.2.3) UNRELEASED; urgency=low
```
Comment (panpan0000) [inline debian/changelog]: per Alan, in Debian rules, before code become a "release", here will be UNRELEASED.
when  release , using "dch -r" , here will become unstable, indicate it's released

File: extra/make-deb.sh [hunk 2]
```diff
@@ -28,15 +20,20 @@ if [ -z "$DEBEMAIL" ]; then
...
 if [ -z "$DEBPKGVER" ]; then
   export DEBPKGVER=`git log -n 1 --pretty=oneline --abbrev-commit`
 fi
 
 if [ -z "$DCHOPTS" ]; then
-        export DCHOPTS="-l ${DEBBRANCH} -u low ${DEBPKGVER}"
+        if [[ $DEBBRANCH =~ ^[0-9.]+$ ]]; then
+                export DCHOPTS="-r \"\""
+        else
+                export DCHOPTS="-v ${DEBBRANCH} -u low ${DEBPKGVER} -b"
```
Comment (panpan0000) [inline extra/make-deb.sh]: please add comment for reason why using "-b" (force build)

File: extra/make-cfg.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+#  This script will modify .bintray-deb.json for deployment on bintray 
+
+# Ensure we're always in the right directory.
+SCRIPT_DIR="$( cd "$( dirname "$0" )" && pwd )"
+cd $SCRIPT_DIR/..
+
+#modify bintray cfg
+DEBBRANCH=$(./extra/gen-debbranch.sh)
+if [[ $DEBBRANCH =~ ^[0-9.]+$ ]]; then
```
Comment (panpan0000) [inline extra/make-cfg.sh]: [minor] it works. but this implementation depends on  how another file extra/gen-debbranch.sh implements 
" if [ "$SYMBOL" != "release" ]; then DEBBRANCH=xxxx"

say, if extra/gen-debbranch.sh changes the "~ ^[0-9.]+$ ", it tends to miss to change this file as well ...
do we have better way , to decouple ?

**Comment by rackhdmirror** [comment]
Build finished.

**Comment by rackhd-test-rel** [comment]
:+1:

**Comment by rackhdmirror** [comment]
Build finished.

**Comment by changev** [comment]
test this please

**Comment by rackhdmirror** [comment]
Build finished.

**Comment by rackhdmirror** [comment]
Build finished.

**Comment by rackhdmirror** [comment]
Build finished.

**Comment by rackhdmirror** [comment]
Build finished.

**Comment by rackhdmirror** [comment]
Build finished.

**Comment by rackhdmirror** [comment]
Build finished.

**Comment by rackhdmirror** [comment]
Build finished.

**Comment by rackhdmirror** [comment]
Build finished.

**Comment by rackhdmirror** [comment]
Build finished.

**Comment by rackhdmirror** [comment]
Build finished.

**Comment by rackhdmirror** [comment]
Build finished.

**Comment by rackhdmirror** [comment]
Build finished.

**Comment by changev** [comment]
test this please

**Comment by rackhdmirror** [comment]
Build finished.

**Comment by rackhdmirror** [comment]
Build finished.

**Comment by rackhdmirror** [comment]
Build finished.

**Pull request closed without merging**